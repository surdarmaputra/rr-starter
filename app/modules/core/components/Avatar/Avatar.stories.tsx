import type { Meta, StoryObj } from '@storybook/react';

import { Avatar, AvatarFallback, AvatarImage } from './Avatar';

type Story = StoryObj<typeof Avatar>;

const meta: Meta = {
  component: Avatar,
  title: 'Core/Components/Avatar',
};

export default meta;

export const Default: Story = {
  render: () => (
    <Avatar data-testid="avatar-testid">
      <AvatarImage alt="@shadcn" src="https://github.com/shadcn.png" />
      <AvatarFallback>CN</AvatarFallback>
    </Avatar>
  ),
};
