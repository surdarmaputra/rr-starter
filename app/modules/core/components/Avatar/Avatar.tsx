import * as AvatarPrimitive from '@radix-ui/react-avatar';
import * as React from 'react';

import { mergeClassNames } from '~/modules/core/libs/utils/ui';

const Avatar = React.forwardRef<
  React.ComponentRef<typeof AvatarPrimitive.Root>,
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Root>
>(({ className, ...props }, ref) => (
  <AvatarPrimitive.Root
    className={mergeClassNames(
      'relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full',
      className,
    )}
    ref={ref}
    {...props}
  />
));
Avatar.displayName = AvatarPrimitive.Root.displayName;

const AvatarImage = React.forwardRef<
  React.ComponentRef<typeof AvatarPrimitive.Image>,
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Image>
>(({ className, ...props }, ref) => (
  <AvatarPrimitive.Image
    className={mergeClassNames('aspect-square h-full w-full', className)}
    ref={ref}
    {...props}
  />
));
AvatarImage.displayName = AvatarPrimitive.Image.displayName;

const AvatarFallback = React.forwardRef<
  React.ComponentRef<typeof AvatarPrimitive.Fallback>,
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Fallback>
>(({ className, ...props }, ref) => (
  <AvatarPrimitive.Fallback
    className={mergeClassNames(
      'flex h-full w-full items-center justify-center rounded-full bg-accent font-extrabold text-letter-title',
      className,
    )}
    ref={ref}
    {...props}
  />
));
AvatarFallback.displayName = AvatarPrimitive.Fallback.displayName;

export { Avatar, AvatarFallback, AvatarImage };
