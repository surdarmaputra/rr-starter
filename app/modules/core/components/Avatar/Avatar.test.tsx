import { render, screen } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import { Avatar, AvatarFallback, AvatarImage } from './Avatar';

describe('Avatar', () => {
  function renderComponent() {
    return render(
      <Avatar data-testid="avatar-testid">
        <AvatarImage alt="@shadcn" src="https://github.com/shadcn.png" />
        <AvatarFallback>CN</AvatarFallback>
      </Avatar>,
    );
  }

  test('renders with data-testid', async () => {
    renderComponent();
    await screen.findByTestId('avatar-testid');
  });

  test('renders snapshot', async () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
