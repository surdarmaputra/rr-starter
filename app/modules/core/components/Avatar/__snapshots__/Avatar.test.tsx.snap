// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > renders snapshot 1`] = `
<div>
  <span
    class="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
    data-testid="avatar-testid"
  >
    <span
      class="flex h-full w-full items-center justify-center rounded-full bg-accent font-extrabold text-letter-title"
    >
      CN
    </span>
  </span>
</div>
`;
